anyway/flask_app.py

```diff
@@ -1648,3 +1627,113 @@ def embedded_reports_api():
     response = Response(json.dumps(embedded_reports_list, default=str), mimetype="application/json")
     response.headers.add("Access-Control-Allow-Origin", "*")
     return response
+
+
+@lm.user_loader
+def load_user(id):
+    return db.session.query(UserOAuth).get(id)
+
+
+@app.route("/user/have_email")
+def user_have_email() -> Response:
+    if current_user.is_anonymous:
+        return Response(
+            "User not login.",
+            status=401,
+        )
+    return jsonify({"have_email": bool(get_current_user_email())})
+
+
+# This code is also used as part of the user first registration
+@app.route("/user/update", methods=["POST"])
+def user_update() -> Response:
+    allowed_fields = [
+        "first_name",
+        "last_name",
+        "email",
+        "phone",
+        "user_type",
+        "user_work_place",
+        "user_url",
+        "user_desc",
+    ]
+
+    if current_user.is_anonymous:
+        return Response(
+            "User not login.",
+            status=401,
+        )
+
+    reg_dict = request.json
+    if not reg_dict:
+        return Response(
+            "Bad response(not a JSON or mimetype does not indicate JSON).",
```

Comment:
Why "response"? Also, why say something like "bad response" if the status already says that?

Reply:
About "response" - it's a copy-paste typo, will fix it.
About why "bad response" - If the user is getting this through a browser then he will not see the HTTP code status.